lib/index.js

```diff
@@ -44,53 +44,113 @@ function PrismPlugin(opts) {
     opts.onlyIn = opts.onlyIn || defaultOnlyIn;
     opts.getSyntax = opts.getSyntax || defaultGetSyntax;
     opts.renderToken = opts.renderToken || defaultTokenRender;
+    opts.blockPerLine = opts.blockPerLine || false;
+
+    let previousBlock;
+    let previousTokens;
 
     function decorate(text, block) {
         let characters = text.characters.asMutable();
-        const string = text.text;
         const grammarName = opts.getSyntax(block);
         const grammar = Prism.languages[grammarName];
+        let tokens;
 
         if (!grammar) {
             return text.characters;
         }
 
-        const tokens = Prism.tokenize(string, grammar);
-        let offset = 0;
+        if (opts.blockPerLine && block === previousBlock) {
```

Comment:
Caching the last computed tokens is a nice optimization. Improves the first decoration pass.
I'll add it to my PR